salt/modules/boto_cloudfront.py

```diff
@@ -53,10 +53,12 @@
 # Import Python libs
 from __future__ import absolute_import, print_function, unicode_literals
 import logging
+import jmespath
```

Comment:
Eh? Should we be conditionally importing this? AFAIK, this won't be present in most installations.

Reply:
Hrm.  It seems to be there on most modern-ish python installs but I see your point.  Ideally we'd add it to the salt.ext bundle, since it's just crazy useful in general, but until that happens I can pretty easily translate the several jmespath calls I make to `xxx.get('foo', {}).get('bar', '')` type calls instead :)  That's what they originally were in my code before I transmuted the to jmespath calls anyway...

Reply:
@cachedout  Considering that ALL the AWS API talks is JSON back and forth, having a lightweight but super convenient library on hand to easily dig around inside that JSON payload (rather than the clumsy .get().get().get() style used so often, or even worse, unsafe assumptions like foo['bar'][0]['baz']) would be a HUGE win for everyone writing boto based modules, and provide a standard approach to such things as well...

How would I go about asking for jmespath to be added to the salt.ext module exactly?  :)

Reply:
We'd have to take our case to @thatch45, @twangboy  and @dmurphy18 and see what they think. The biggest issue here is packaging, which is a pretty major pain. We'd much rather see this as an optional dependency but let's see what the feedback looks like.

Reply:
I can definitely make it optional, but that just ends up with two code-paths inside each function - one using JMESPath, and one using raw python (generally something on the order of `xxx.get('foo', {}).get('bar', {})[0].get('baz')` or such crappiness.  If we're going to have to put the latter into each function anyway, there's zero advantage to having the alternate JMESPath version to begin with :)

Fingers crossed ...

Reply:
@cachedout It is generally available, e.g. Redhat extras repo, so so long as expect extras to be available, then updating salt requires should be easy enough. Question is would this be from Neon forward, or backport to Fluorine ?

Reply:
I have no particular need to backport it right this moment, since I already have JMESPath installed on my salt masters (due to the other modules such as boto_rds which require it).  That said, there ARE the several modules mentioned below which already use it, so if nothing else, adding it to the requires for Fluorine would help allay any confusion which might otherwise arise if someone happened to be missing it, but tried out those modules :)

Reply:
@cachedout the jmespath module installed cleanly on Windows when I just tried it. Just let me know if I need to add it to the deps for Windows.